src/encode.rs

```diff
@@ -0,0 +1,296 @@
+use crate::{
+    errors::{Error, Result},
+    key::PublicKeyParts,
+    parse::rsa_oid,
+    RSAPrivateKey, RSAPublicKey,
+};
+use num_bigint::{BigUint, ToBigInt};
+use num_traits::Zero;
+use simple_asn1::{to_der, ASN1Block};
+
+#[cfg(feature = "pem")]
```

Comment:
This could then be `trait PrivateKeyPemEncoding : PrivateKeyEncoding` with auto-implementations, as its methods only rely on the `PrivateKeyEncoding` API.